guillotina_amqp/job.py

```diff
@@ -94,42 +88,16 @@ def state_manager(self):
         return self._state_manager
 
     async def create_request(self):
-        req_data = self.data["req_data"]
-        url = req_data["url"]
-        parsed = urlparse(url)
-        dct = {
-            "method": req_data["method"],
-            "url": yarl.URL(url),
-            "path": parsed.path,
-            "headers": CIMultiDict(req_data["headers"]),
-            "raw_headers": tuple(
-                (k.encode("utf-8"), v.encode("utf-8"))
-                for k, v in req_data["headers"].items()
-            ),
-        }
-
-        message = self.base_request._message._replace(**dct)
-
-        payload_writer = mock.Mock()
-        payload_writer.write_eof.side_effect = noop
-        payload_writer.drain.side_effect = noop
-
-        protocol = mock.Mock()
-        protocol.transport = test_utils._create_transport(None)
-        protocol.writer = payload_writer
-
         request = self.base_request.__class__(
-            message,
-            EmptyPayload(),
-            protocol,
-            payload_writer,
-            self.task,
-            self.task._loop,
+            self.base_request.scheme,
+            self.base_request.method,
+            self.base_request.path,
+            self.base_request.query_string.encode("utf-8"),
+            self.base_request.raw_headers.copy(),
             client_max_size=self.base_request._client_max_size,
-            state=self.base_request._state.copy(),
         )
+        request._state = self.base_request._state.copy()
```

Comment:
Can be dangerous modify this private attribute?

Reply:
yes! probably guillotina has already a util to clone a request without having to do it in a "dirty" way like here.

Reply:
@vangheem any hints/ideas on how to improve this part? I would prefer this to be done as a separate PR though

Reply:
hmmm, I wonder if we don't need this anymore.

I imagine this was from back in the python asyncio 3.5 days where we were putting everything on the request object. Since guillotina shouldn't depend on request object anymore and all we need to know about is authz, maybe we can change this... but maybe not this pr.

How else do you want to improve it though? Maybe https://github.com/plone/guillotina/blob/master/guillotina/request.py#L279 needs a clone operation?

Reply:
Opened a separate ticket to improve this: https://github.com/guillotinaweb/guillotina_amqp/issues/52